.github/workflows/continuous-integration.yml

```diff
@@ -322,12 +324,62 @@ jobs:
 
       - run: cp .env.ci .env
 
-      - run: docker-compose up -d
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v1
 
+      # build API (using cache; provide image to docker-compose)
+      - name: Build docker image (API/PHP)
+        uses: docker/build-push-action@v2
+        with:
+          file: api/Dockerfile
+          context: './api'
+          push: false
+          load: true
+          target: api_platform_php_dev
+          builder: ${{ steps.buildx.outputs.name }}
+          tags: ecamp/ecamp3-api
+          cache-from: type=gha,scope=api
+          cache-to: type=gha,scope=api,mode=max
+
+      # build caddy (using cache; provide image to docker-compose)
+      - name: Build docker image (Caddy)
+        uses: docker/build-push-action@v2
+        with:
+          file: api/Dockerfile
+          context: './api'
+          push: false
+          load: true
+          target: api_platform_caddy
+          builder: ${{ steps.buildx.outputs.name }}
+          tags: ecamp/ecamp3-caddy
+          cache-from: type=gha,scope=caddy
+          cache-to: type=gha,scope=caddy,mode=max
```

Comment:
Der Prozess könnte noch weiter beschleunigt werden, wenn du diese drei Steps stattdessen als separate Jobs definierst (mit den entsprechenden `needs:` Angaben). So könnten nämlich die zwei Docker image builds noch parallel laufen. Da sich am Caddy selten etwas ändern wird ists wohl meistens auch nicht viel langsamer so wie dus jetzt gemacht hast. Aber wenn wir mal noch mehr als nur diese zwei Images könnte sich das zu lohnen beginnen.

Reply:
Im Moment würd ich das lieber so lassen. Ich musste so schon ziemlich herumprobieren, damit docker-compose die images findet/liest, welche vorher durch buildx gebaut wurden.

Wenn ich das builden in einen anderen job auslagere, dann muss ich dass irgendwie wieder im cache zwischenspeichern. Wüsste ich jetzt nicht grad auf Anhieb wie.